src/safeposix/syscalls/net_calls.rs

```diff
@@ -3084,90 +3084,193 @@ impl Cage {
                 let sock_tmp = sockfdobj.handle.clone();
                 let mut sockhandle = sock_tmp.write();
                 match level {
+                    // a few UDP options are avaliable in Linux
+                    // though we do not support them for now
                     SOL_UDP => {
                         return syscall_error(
-                            Errno::EOPNOTSUPP,
+                            Errno::ENOPROTOOPT,
                             "getsockopt",
                             "UDP is not supported for getsockopt",
                         );
                     }
                     SOL_TCP => {
                         // Checking the tcp_options here
                         // Currently only support TCP_NODELAY option for SOL_TCP
+
+                        // TCP_NODELAY: If set, disable the Nagle algorithm. This means that
+                        // segments are always sent as soon as possible, even if
+                        // there is only a small amount of data. When not set, data
+                        // is buffered until there is a sufficient amount to send
+                        // out, thereby avoiding the frequent sending of small
+                        // packets, which results in poor utilization of the network.
+                        // This option is overridden by TCP_CORK; however, setting
+                        // this option forces an explicit flush of pending output,
+                        // even if TCP_CORK is currently set.
                         if optname == TCP_NODELAY {
                             let optbit = 1 << optname;
                             if optbit & sockhandle.tcp_options == optbit {
+                                // if the bit is set, set optval to 1
                                 *optval = 1;
                             } else {
+                                // otherwise, set optval to 0
                                 *optval = 0;
                             }
                             return 0;
                         }
+                        // other TCP options are not supported yet
                         return syscall_error(
-                            Errno::EOPNOTSUPP,
+                            Errno::ENOPROTOOPT,
                             "getsockopt",
                             "TCP options not remembered by getsockopt",
                         );
                     }
                     SOL_SOCKET => {
                         // checking the socket_options here
                         match optname {
-                            //indicate whether we are accepting connections or not in the moment
+                            // indicate whether we are accepting connections or not in the moment
                             SO_ACCEPTCONN => {
                                 if sockhandle.state == ConnState::LISTEN {
+                                    // if in LISTEN state, set optval to 1
```

Comment:
optval is the return value. SO_ACCEPTCONN indicates whether we are accepting connections or not in the moment as mentioned in the comment, so we just check if current state is LISTEN. If the state is listen, we set return value to 1, otherwise 0